src/python/dart/service/workflow.py

```diff
@@ -157,14 +157,35 @@ def _query_workflow_instance_query(self, filters):
             query = self._filter_service.apply_filter(f, query, WorkflowInstanceDao, [workflow_instance_schema()])
         return query
 
-    def action_checkout(self, action):
-        action = self._action_service.update_action_state(action, ActionState.RUNNING, action.data.error_message)
-        if action.data.workflow_instance_id and action.data.first_in_workflow:
-            wf_instance = self.get_workflow_instance(action.data.workflow_instance_id)
-            self.update_workflow_instance_state(wf_instance, WorkflowInstanceState.RUNNING)
-            wf = self.get_workflow(action.data.workflow_id)
-            if wf.data.on_started_email:
-                self._emailer.send_workflow_started_email(wf, wf_instance)
+    def action_checkout(self, input_action):
+        action = None
+        wf_instance = None
+        step = "Starting action checkout"
+        try:
+            action = self._action_service.update_action_state(input_action, ActionState.RUNNING, input_action.data.error_message)
+            step = "Updated action state to running"
+
+            if action.data.workflow_instance_id and action.data.first_in_workflow:
+                wf_instance = self.get_workflow_instance(action.data.workflow_instance_id)
+                step = "Got workflow instance {0} since it is the first action in workflow".format(action.data.workflow_instance_id)
```

Comment:
I do not think there is a  pythonic preference for the {} usage. But the use of '%' is now considered outdated.